beacon-chain/sync/validate_beacon_attestation.go

```diff
@@ -240,29 +233,25 @@ func (s *Service) validateCommitteeIndex(
 	a eth.Att,
 	bs state.ReadOnlyBeaconState,
 ) (primitives.CommitteeIndex, uint64, pubsub.ValidationResult, error) {
-	valCount, err := helpers.ActiveValidatorCount(ctx, bs, slots.ToEpoch(a.GetData().Slot))
-	if err != nil {
-		return 0, 0, pubsub.ValidationIgnore, err
-	}
-	count := helpers.SlotCommitteeCount(valCount)
-	if uint64(a.GetData().CommitteeIndex) > count {
-		return 0, 0, pubsub.ValidationReject, errors.Errorf("committee index %d > %d", a.GetData().CommitteeIndex, count)
-	}
-
 	var ci primitives.CommitteeIndex
+	var pb pubsub.ValidationResult
+	var err error
 	if a.Version() >= version.Electra {
-		dataCi := a.GetData().CommitteeIndex
-		if dataCi != 0 {
-			return 0, 0, pubsub.ValidationReject, fmt.Errorf("committee index must be 0 but was %d", dataCi)
+		ci, pb, err = validateCommitteeIndexElectra(ctx, a)
+		if err != nil {
+			return 0, 0, pb, errors.Wrap(err, "failed to validate committee index")
 		}
-		indices := helpers.CommitteeIndices(a.CommitteeBitsVal())
-		if len(indices) != 1 {
-			return 0, 0, pubsub.ValidationReject, fmt.Errorf("exactly 1 committee index must be set but %d were set", len(indices))
-		}
-		ci = indices[0]
 	} else {
 		ci = a.GetData().CommitteeIndex
 	}
+	valCount, err := helpers.ActiveValidatorCount(ctx, bs, slots.ToEpoch(a.GetData().Slot))
+	if err != nil {
+		return 0, 0, pubsub.ValidationIgnore, err
+	}
+	count := helpers.SlotCommitteeCount(valCount)
+	if uint64(ci) > count {
```

Comment:
sometimes we need this check and other times we don't need it, I wonder why the difference